File: day 0 run length encoding/RunLengthEncoding.java
import java.util.Scanner;
class RunLengthEncoding{
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in);
		String str=sc.next()+"*";
		int count=0;
		for(int i=0;i<str.length()-1;i++){
			count++;
			if(str.charAt(i)!=str.charAt(i+1)){
				System.out.print(str.charAt(i)+""+count);
				count=0;
			}
		}
	}
}

File: day 0 run length encoding/run length encoding.c
#include<stdio.h>

int main(){
	char str[101];
	int len=0;
	scanf("%s%n",str,&len);
	str[len]='*';
	str[len]='\0';
	int count=1;
	for(int i=0;i<len;i++){
		if(str[i]!=str[i+1]){
			printf("%c%d",str[i],count);
			count=1;
			continue;
		}
		count++;
	}
	return 0;
}
